components/automate-cli/cmd/chef-automate/restart.go

```diff
@@ -49,6 +97,250 @@ func runRestartServices(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(restartServicesCmd)
+type restartFromBastion interface {
+	getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error)
+	isManagedServicesOn() bool
+	executeRemoteExecutor(*NodeTypeAndCmd, SSHUtil, *cli.Writer) (map[string][]*CmdResult, error)
+	printRestartCmdOutput(map[string][]*CmdResult, string, *sync.WaitGroup, *sync.Mutex, *cli.Writer)
+}
+
+type restartFromBastionImpl struct{}
+
+func NewRestartFromBastionImpl() *restartFromBastionImpl {
+	return &restartFromBastionImpl{}
+}
+
+func (rs *restartFromBastionImpl) getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error) {
+	return getAutomateHAInfraDetails()
+}
+
+func (st *restartFromBastionImpl) isManagedServicesOn() bool {
+	return isManagedServicesOn()
+}
+
+func (rs *restartFromBastionImpl) executeRemoteExecutor(nodemap *NodeTypeAndCmd, sshUtil SSHUtil, writer *cli.Writer) (map[string][]*CmdResult, error) {
+	remoteExecutor := NewRemoteCmdExecutor(nodemap, sshUtil, writer)
+	cmdResult, err := remoteExecutor.Execute()
+	return cmdResult, err
+}
+
+func (rs *restartFromBastionImpl) printRestartCmdOutput(cmdResult map[string][]*CmdResult, remoteService string, wg *sync.WaitGroup, mutex *sync.Mutex, writer *cli.Writer) {
+	mutex.Lock()
+	writer.Printf("=====================================================\n")
+	for _, value := range cmdResult {
+		for _, cmdResult := range value {
+			printOutput(remoteService, *cmdResult, []string{}, writer)
+		}
+	}
+	mutex.Unlock()
+	wg.Done()
+}
+
+type MockrestartFromBastionImpl struct {
+	getAutomateHAInfraDetailsFunc func() (*AutomateHAInfraDetails, error)
+	isManagedServicesOnFunc       func() bool
+	executeRemoteExecutorFunc     func(*NodeTypeAndCmd, SSHUtil, *cli.Writer) (map[string][]*CmdResult, error)
+	printRestartCmdOutputFunc     func(map[string][]*CmdResult, string, *sync.WaitGroup, *sync.Mutex, *cli.Writer)
+}
+
+func (mrs *MockrestartFromBastionImpl) getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error) {
+	return mrs.getAutomateHAInfraDetailsFunc()
+}
+
+func (mrs *MockrestartFromBastionImpl) isManagedServicesOn() bool {
+	return mrs.isManagedServicesOnFunc()
+}
+
+func (mrs *MockrestartFromBastionImpl) executeRemoteExecutor(nodemap *NodeTypeAndCmd, sshUtil SSHUtil, writer *cli.Writer) (map[string][]*CmdResult, error) {
+	return mrs.executeRemoteExecutorFunc(nodemap, sshUtil, writer)
+}
+
+func (mrs *MockrestartFromBastionImpl) printRestartCmdOutput(cmdResult map[string][]*CmdResult, remoteService string, wg *sync.WaitGroup, mutex *sync.Mutex, writer *cli.Writer) {
+	mrs.printRestartCmdOutputFunc(cmdResult, remoteService, wg, mutex, writer)
+}
+
+func runRestartFromBastion(flags *RestartCmdFlags, rs restartFromBastion) error {
```

Comment:
Break this function with fewer Cognitive Complexity to avoid code smell

Reply:
Done